modules/common/types/PatternParams.ts

```diff
@@ -0,0 +1,49 @@
+import { Texture, Vector2 } from "three";
+
+export type PatternParam<T = Vector2 | number | Texture> = {
+  name: string;
+  value: T;
+};
+export type PatternParams = {
+  [key: string]: PatternParam;
+};
```

Comment:
```suggestion
export type PatternParams = Record<string, PatternParam>;
```

Reply:
Wouldn't let me merge your suggestion 🤷 but incorporated it under my name :imadethis: